packages/cli/src/rpc/navie/isCustomWelcomeMessageEnabled.ts

```diff
@@ -1,5 +1,18 @@
+import ModelRegistry from './models/registry';
+
 export default function isCustomWelcomeMessageEnabled(): boolean {
   // This behavior is DEFAULT ON, but can be disabled by setting the environment variable
-  // NAVIE_CUSTOM_WELCOME_MESSAGE=false. Any other value will be treated as true.
-  return process.env.NAVIE_CUSTOM_WELCOME_MESSAGE !== 'false';
+  // NAVIE_CUSTOM_WELCOME_MESSAGE=false.
+  if (process.env.NAVIE_CUSTOM_WELCOME_MESSAGE === 'false') {
+    return false;
+  }
+
+  // If the model selector is available, the custom message can only be rendered if a model is
+  // selected.
+  if (process.env.APPMAP_NAVIE_MODEL_SELECTOR) {
+    return ModelRegistry.instance.selectedModel !== undefined;
+  }
```

Comment:
This `APPMAP_NAVIE_MODEL_SELECTOR` environment variable controls the behavior of the welcome message. If the model selector is available, this environment variable should be set. When it is set, it prevents the welcome message from attempting to generate completions unless a model is actively selected.